### PR TITLE
fix(patch_parsing): lines starting with backslash

### PR DESCRIPTION
**Pull request by sapristi**
Handles bug encountered due to the presence of `\ No newline at end of file` in patches, that were not handled

### CODE REVIEW

File: ggshield/utils.py [hunk 1]
```diff
@@ -147,6 +147,8 @@ def get_lines_from_patch(content: str, filemode: Filemode) -> Iterable[Line]:
             line_pre_index = pre_index
             line_content = line[1:]
             category = LineCategory.deletion
+        elif line_type == "\\":
+            line_content = line[1:]
```
Comment (agateau-gg) [inline ggshield/utils.py:151]: This makes sense, but can you add a test for it?

Comment (sapristi) [reply ggshield/utils.py:151]: Added a supplementary test case for the `make_indices`.
There is no test yet for the `get_lines_from_file` function.

Comment (agateau-gg) [reply ggshield/utils.py:151]: Looks good to me :+1:.

**Comment by sapristi** [comment]
@jeremyds can you review and merge if appropriate ?

File: ggshield/utils.py (hunk 1 above)
Comment (agateau-gg) [reply ggshield/utils.py:151]: Actually something is odd: codecov says [your code is not reached by the test suite](https://app.codecov.io/gh/GitGuardian/ggshield/compare/137/diff?src=pr&el=tree). Can you look into this?

Comment (sapristi) [reply ggshield/utils.py:151]: Seems fixed with latest changes

**Review by agateau-gg — APPROVED** [review]

**Pull request merged**